pkg/virt-launcher/virtwrap/nichotplug.go

```diff
@@ -36,67 +36,63 @@ import (
+		if err := vmConfigurator.SetupPodNetworkPhase2(updatedDomain, []v1.Network{*network}); err != nil {
 			return err
 		}
 
-		relevantIface := domainInterfaceFromNetwork(updatedDomain, network)
+		relevantIface := lookupInterfaceName(currentDomain.Spec.Devices.Interfaces, networkName)
 		if relevantIface == nil {
-			return fmt.Errorf("could not retrieve the api.Interface object from the dummy domain")
+			return fmt.Errorf("could not retrieve %q Interface object from domain", networkName)
 		}
 
-		log.Log.Infof("will hot plug %s with MAC %s", network.Name, *relevantIface.MAC)
+		log.Log.Infof("will hot plug %s with MAC %s", networkName, relevantIface.MAC)
 		ifaceXML, err := xml.Marshal(relevantIface)
 		if err != nil {
 			return err
 		}
 
 		if err := dom.AttachDeviceFlags(strings.ToLower(string(ifaceXML)), affectLiveAndConfigLibvirtFlags); err != nil {
-			log.Log.Reason(err).Errorf("libvirt failed to attach interface %s: %v", network.Name, err)
+			log.Log.Reason(err).Errorf("libvirt failed to attach interface %s: %v", networkName, err)
 			return err
 		}
 	}
 	return nil
 }
 
-func domainInterfaceFromNetwork(domain *api.Domain, network v1.Network) *api.Interface {
-	for _, iface := range domain.Spec.Devices.Interfaces {
-		if iface.Alias.GetName() == network.Name {
-			return &iface
-		}
-	}
-	return nil
-}
-
-func networksToHotplugWhoseInterfacesAreNotInTheDomain(vmi *v1.VirtualMachineInstance, indexedDomainIfaces map[string]api.Interface) []v1.Network {
-	var networksToHotplug []v1.Network
-	interfacesToHoplug := netvmispec.IndexInterfacesFromStatus(
-		vmi.Status.Interfaces,
-		func(ifaceStatus v1.VirtualMachineInstanceNetworkInterface) bool {
-			_, exists := indexedDomainIfaces[ifaceStatus.Name]
-
-			return netvmispec.ContainsInfoSource(
-				ifaceStatus.InfoSource, netvmispec.InfoSourceMultusStatus,
-			) && !exists
-		},
-	)
+func networksToHotplugWhoseInterfacesAreNotInTheDomain(vmiStatusInterfaces []v1.VirtualMachineInstanceNetworkInterface, domainInterfaces []api.Interface) []string {
+	var networks []string
+	lookupDomainIfaceWithName := indexedDomainInterfaces(domainInterfaces)
+	for _, vmiStatusIface := range vmiStatusInterfaces {
+		_, ifaceAttachedToDomain := lookupDomainIfaceWithName[vmiStatusIface.Name]
+		ifaceExistInPod := netvmispec.ContainsInfoSource(vmiStatusIface.InfoSource, netvmispec.InfoSourceMultusStatus)
 
-	for netName, network := range netvmispec.IndexNetworkSpecByName(vmi.Spec.Networks) {
-		if _, isAttachmentToBeHotplugged := interfacesToHoplug[netName]; isAttachmentToBeHotplugged {
-			networksToHotplug = append(networksToHotplug, network)
+		if ifaceExistInPod && !ifaceAttachedToDomain {
+			networks = append(networks, vmiStatusIface.Name)
 		}
 	}
 
-	return networksToHotplug
+	return networks
 }
 
-func indexedDomainInterfaces(domain *api.Domain) map[string]api.Interface {
+func indexedDomainInterfaces(ifaces []api.Interface) map[string]api.Interface {
```

Comment:
makes sense; do you mind updating the func name as well, to `indexDomainInterfaces` ? 

That would allow you variable in https://github.com/kubevirt/kubevirt/pull/9538/files#diff-a4d56356c36cdb90f1e866f87d3d84776fffe23bf5dbe1cc89d5b36dd94e4454R68 to be named `indexedDomainInterfaces`, which I prefer. I don't understand the reasoning behind the current name you've suggested.